File: python/test/test_transform.py
import ThrustRTC as trtc



is_odd = trtc.Functor( {}, ['x'], 
'''
         return x % 2;
''')

darr = trtc.device_vector_from_list([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8], 'int32_t')
trtc.Transform(darr, darr, trtc.Negate())
print (darr.to_host())

darr_in1 = trtc.device_vector_from_list([-5,  0,  2,  3,  2,  4 ], 'int32_t')
darr_in2 = trtc.device_vector_from_list([ 3,  6, -2,  1,  2,  3 ], 'int32_t')
darr_out = trtc.device_vector('int32_t', 6)
trtc.Transform_Binary(darr_in1, darr_in2, darr_out, trtc.Plus())
print (darr_out.to_host())

darr = trtc.device_vector_from_list([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8], 'int32_t')
trtc.Transform_If(darr, darr, trtc.Negate(), is_odd)
print (darr.to_host())

darr_data = trtc.device_vector_from_list([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8 ], 'int32_t')
darr_stencil = trtc.device_vector_from_list([1, 0, 1,  0, 1, 0, 1,  0, 1, 0], 'int32_t')
trtc.Transform_If_Stencil(darr_data, darr_stencil, darr_data, trtc.Negate(), trtc.Identity())
print (darr_data.to_host())

darr_in1 = trtc.device_vector_from_list([-5,  0,  2,  3,  2,  4 ], 'int32_t')
darr_in2 = trtc.device_vector_from_list([ 3,  6, -2,  1,  2,  3 ], 'int32_t')
darr_stencil = trtc.device_vector_from_list([ 1,  0,  1,  0,  1,  0], 'int32_t')
darr_out =  trtc.device_vector_from_list([ -1, -1, -1, -1, -1, -1 ], 'int32_t')
trtc.Transform_Binary_If_Stencil(darr_in1, darr_in2, darr_stencil, darr_out, trtc.Plus(), trtc.Identity())
print (darr_out.to_host())

File: java/test/test_transform.java
import java.util.*;
import JThrustRTC.*;

public class test_transform
{
	public static void main(String[] args) 
	{	
        Functor is_odd = new Functor(new String[] { "x" }, "        return x % 2;\n");

        {
            DVVector vec = new DVVector(new int[] { -5, 0, 2, -3, 2, 4, 0, -1, 2, 8 });
            TRTC.Transform(vec, vec, new Functor("Negate")); // in place
        	System.out.println(Arrays.toString((int[])vec.to_host()));
        }

        {
            DVVector d_in1 = new DVVector(new int[] { -5, 0, 2, 3, 2, 4 });
            DVVector d_in2 = new DVVector(new int[] { 3, 6, -2, 1, 2, 3 });
            DVVector d_out = new DVVector("int32_t", 6);
            TRTC.Transform_Binary(d_in1, d_in2, d_out, new Functor("Plus"));
            System.out.println(Arrays.toString((int[])d_out.to_host()));
        }

        {
            DVVector vec = new DVVector(new int[] { -5, 0, 2, -3, 2, 4, 0, -1, 2, 8 });
            TRTC.Transform_If(vec, vec, new Functor("Negate"), is_odd); // in place
            System.out.println(Arrays.toString((int[])vec.to_host()));
        }

        {
            DVVector d_data = new DVVector(new int[] { -5, 0, 2, -3, 2, 4, 0, -1, 2, 8 });
            DVVector d_stencil = new DVVector(new int[] { 1, 0, 1, 0, 1, 0, 1, 0, 1, 0 });
            TRTC.Transform_If_Stencil(d_data, d_stencil, d_data, new Functor("Negate"), new Functor("Identity")); // in place
            System.out.println(Arrays.toString((int[])d_data.to_host()));
        }

        {
            DVVector d_in1 = new DVVector(new int[] { -5, 0, 2, 3, 2, 4 });
            DVVector d_in2 = new DVVector(new int[] { 3, 6, -2, 1, 2, 3 });
            DVVector d_stencil = new DVVector(new int[] { 1, 0, 1, 0, 1, 0 });
            DVVector d_output = new DVVector("int32_t", 6);
            TRTC.Transform_Binary_If_Stencil(d_in1, d_in2, d_stencil, d_output, new Functor("Plus"), new Functor("Identity")); // in place
            System.out.println(Arrays.toString((int[])d_output.to_host()));
        }

	}	

}
